specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2021-07-22-preview/PrometheusRuleGroups.json

```diff
@@ -0,0 +1,601 @@
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PrometheusRuleGroupProperties",
+          "description": "The alert rule properties of the resource."
+        }
+      },
+      "description": "The Prometheus rule group resource for patch operations."
+    },
+    "PrometheusRuleGroupResourceCollection": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrometheusRuleGroupResource"
+          },
+          "description": "the values for the alert rule resources."
+        }
+      },
+      "description": "Represents a collection of alert rule resources."
+    },
+    "PrometheusRule": {
+      "type": "object",
+      "properties": {
+        "record": {
+          "type": "string",
+          "description": "the name of the recording rule."
+        },
+        "alert": {
+          "type": "string",
+          "description": "the name of the alert rule."
+        },
+        "expression": {
+          "type": "string",
+          "description": "the expression to run for the rule."
+        },
+        "severity": {
+          "type": "integer",
+          "format": "int32",
+          "description": "the expression to run for the rule."
+        },
+        "for": {
+          "type": "string",
+          "description": "the amount of time alert must be active before firing."
+        },
+        "labels": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "labels for rule group"
+        },
+        "annotations": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "annotations for rule group"
+        },
+        "actions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrometheusRuleGroupAction"
+          },
+          "description": "the array of actions that are performed when the alert rule becomes active, and when an alert condition is resolved."
+        },
+        "resolveConfiguration": {
+          "$ref": "#/definitions/PrometheusRuleResolveConfiguration",
+          "description": "defines the configuration for resolving fired alerts."
+        }
+      }
+    },
+    "PrometheusRuleResolveConfiguration": {
+      "type": "object",
+      "properties": {
+        "autoResolved": {
+          "type": "boolean",
+          "description": "the flag that indicates whether or not to auto resolve a fired alert."
+        },
+        "timeToResolve": {
+          "type": "string",
+          "description": "the duration a rule must evaluate as healthy before the fired alert is automatically resolved."
+        }
+      },
+      "description": "Specifies the Prometheus alert rule configuration."
+    },
+    "PrometheusRuleGroupsErrorResponse": {
```

Comment:
>PrometheusRuleGroupsErrorResponse [](http://example.com/codeflow?start=5&length=33)

Recommend using the ErrorResponse type in common-types:

"$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse".

Reply:
Updated.

Reply:
You should remove this and the next definition since they aren't used any more.

Reply:
Thanks - done.